src/main/java/knightminer/simplytea/item/TeapotItem.java

```diff
@@ -33,40 +40,44 @@ public TeapotItem(Properties props) {
 	public InteractionResultHolder<ItemStack> use(Level world, Player player, InteractionHand hand) {
 		ItemStack stack = player.getItemInHand(hand);
 		BlockHitResult rayTrace = getPlayerPOVHitResult(world, player, ClipContext.Fluid.SOURCE_ONLY);
-		if (rayTrace.getType() == Type.BLOCK) {
-			BlockPos pos = rayTrace.getBlockPos();
-			BlockState state = world.getBlockState(pos);
-
-			// we use name for lookup to prevent default fluid conflicts
-			Fluid fluid = state.getFluidState().getType();
-			if(fluid != Fluids.EMPTY) {
-				// try for water or milk using the config lists
-				Item item = null;
-				if (fluid.is(FluidTags.WATER)) {
-					item = Registration.teapot_water;
-				} // TODO: milk when mods make a standard
-
-				// if either one is found, update the stack
-				if(item != null) {
-					// water is considered infinite unless disabled in the config
-					if(!Config.SERVER.teapot.infiniteWater()) {
-						Direction side = rayTrace.getDirection();
-						// unable to modify the block, fail
-						if (!world.mayInteract(player, pos) || !player.mayUseItemAt(pos.relative(side), side, stack) || !(state.getBlock() instanceof BucketPickup)) {
-							return new InteractionResultHolder<>(InteractionResult.FAIL, stack);
-						}
-						((BucketPickup)state.getBlock()).pickupBlock(world, pos, state);
-					}
+		if (rayTrace.getType() != Type.BLOCK) {
+			return InteractionResultHolder.pass(stack);
+		}
+		
+		BlockPos pos = rayTrace.getBlockPos();
+		Direction side = rayTrace.getDirection();
+		BlockState state = world.getBlockState(pos);
 
-					stack = ItemUtils.createFilledResult(stack, player, new ItemStack(item));
+		if (!world.mayInteract(player, pos) || !player.mayUseItemAt(pos.relative(side), side, stack)) {
+			return InteractionResultHolder.fail(stack);
+		}
 
-					// TODO: fluid sound based on fluid
-					player.playSound(SoundEvents.BUCKET_FILL, 1.0f, 1.0f);
-					return new InteractionResultHolder<>(InteractionResult.SUCCESS, stack);
+		ItemStack filledStack = ItemStack.EMPTY;
+		if (state.getBlock() instanceof BucketPickup bucketPickup) {
+			// special case for infinite water
+			if (FluidTeapotWrapper.isWater(state.getFluidState().getType()) && !Config.SERVER.teapot.infiniteWater()) {
+				filledStack = new ItemStack(Registration.teapot_water);
+				Optional<SoundEvent> sound = bucketPickup.getPickupSound(state);
+				if (sound.isPresent()) {
+					player.playSound(sound.get(), 1.0f, 1.0f);
+				}
+			}
+			
+			// use teapot like a bucket to get fluid, should work in most cases
```

Comment:
Does this work if you make the config lower than 1000 for the teapot capacity? If not we need to special case that here.

Reply:
With the way FluidUtil is handling it, it should be treated as a successful pickup as long as the item's fluid handler returns a non-zero filled amount.

The item fluid handler (FluidTeapotWrapper) uses the configured size. It will either "fill" 0 (fail) or teapotCapacity (success).